discretize/CylMesh.py

```diff
@@ -218,16 +398,276 @@ def vol(self):
+                np.ones(self.nCz, dtype=bool),
+                np.kron(
+                    np.ones(self.nCy, dtype=bool),
+                    hang_x
+                )
+            )
+        return self.__ishangingFx
+
+    @property
+    def _hangingFx(self):
+        if getattr(self, '__hangingFx', None) is None:
+            self.__hangingFx = dict(zip(
+                np.nonzero(self._ishangingFx)[0].tolist(), [None]*self._nhFx
+            ))
+        return self.__hangingFx
+
+    @property
+    def _ishangingFy(self):
+        if getattr(self, '__ishangingFy', None) is None:
+            hang_y = np.zeros(self._ntNy, dtype=bool)
+            hang_y[-1] = True
+            self.__ishangingFy = np.kron(
+                np.ones(self.nCz, dtype=bool),
+                np.kron(
+                    hang_y,
+                    np.ones(self.nCx, dtype=bool)
+                )
+            )
+        return self.__ishangingFy
+
+    @property
+    def _hangingFy(self):
+        if getattr(self, '__hangingFy', None) is None:
+            deflate_y = np.zeros(self._ntNy, dtype=bool)
+            deflate_y[0] = True
+            deflateFy = np.nonzero(np.kron(
+                np.ones(self.nCz, dtype=bool),
+                np.kron(
+                    deflate_y,
+                    np.ones(self.nCx, dtype=bool)
+                )
+            ))[0].tolist()
+            self.__hangingFy = dict(zip(
+                np.nonzero(self._ishangingFy)[0].tolist(),
+                deflateFy)
+            )
+        return self.__hangingFy
+
+    @property
+    def _ishangingFz(self):
+        if getattr(self, '__ishangingFz', None) is None:
+            self.__ishangingFz = np.kron(
+                np.zeros(self.nNz, dtype=bool),
+                np.kron(
+                    np.zeros(self.nCy, dtype=bool),
+                    np.zeros(self.nCx, dtype=bool)
+                )
+            )
+        return self.__ishangingFz
+
+    @property
+    def _hangingFz(self):
+        return {}
+
+    @property
+    def _ishangingEx(self):
+        if getattr(self, '__ishangingEx', None) is None:
+            hang_y = np.zeros(self._ntNy, dtype=bool)
+            hang_y[-1] = True
+            self.__ishangingEx = np.kron(
+                np.ones(self._ntNz, dtype=bool),
+                np.kron(
+                    hang_y,
+                    np.ones(self.nCx, dtype=bool)
+                )
+            )
+        return self.__ishangingEx
+
+    @property
+    def _hangingEx(self):
+        if getattr(self, '__hangingEx', None) is None:
+            deflate_y = np.zeros(self._ntNy, dtype=bool)
+            deflate_y[0] = True
+            deflateEx = np.nonzero(np.kron(
+                np.ones(self._ntNz, dtype=bool),
+                np.kron(
+                    deflate_y,
+                    np.ones(self.nCx, dtype=bool)
+                )
+            ))[0].tolist()
+            self.__hangingEx = dict(zip(
```

Comment:
there is a fair bit of changing of types here
(bool --> ints (indices) --> list --> list of tuples --> dict)

This was the way that was most convenient for me to work with - open to suggestions